src/IntegratedLibrary/fault/MemoryManipulationSystem.cpp

```diff
@@ -0,0 +1,246 @@
+            }
+            arch_width = reader.get_class() == ELFCLASS64 ? 64 : 32;
+        }
+
+        etiss::log(etiss::VERBOSE, std::string("Mounted ") + this->getName() + std::string("'s VirtualStruct to CPUCore ") + vs_parent->getName());
+
+        if (!etiss::cfg().isSet("vp.elf_file"))
+        {
+            etiss::log(etiss::FATALERROR, std::string("MemoryManipulationSystem::initialize_virtualstruct: Requires \"vp.elf_file\" config to retrieve architecture bit-width."));
+        }
+
+        std::string elf_file = etiss::cfg().get<std::string>("vp.elf_file", "");
+
+        ELFIO::elfio reader;
+
+        if (!reader.load(elf_file))
+        {
+            etiss::log(etiss::FATALERROR, "ELF reader could not process file");
+        }
+
+        if (arch_width == 32)
+        {
+            auto read = [this](size_t address, etiss::int32& return_code) {
+              etiss::uint32 x;
+              return_code = dbus_access<false>(nullptr, static_cast<etiss::uint64>(address), reinterpret_cast<etiss::uint8 *>(&x), sizeof(x));
+              return x;
+            };
+            auto write = [this](size_t address, etiss::uint32 word, etiss::int32& return_code) {
+              return_code = dbus_access<true>(nullptr, static_cast<etiss::uint64>(address), reinterpret_cast<etiss::uint8 *>(&word), sizeof(word));
+            };
+            mem_manipulator_ = std::make_unique<MemoryWordManipulator<etiss::uint32>>(read, write);
+        }
+        else if (arch_width == 64)
+        {
+            auto read = [this](size_t address, etiss::int32& return_code) {
+              etiss::uint64 x;
+              return_code = dbus_access<false>(nullptr, static_cast<etiss::uint64>(address), reinterpret_cast<etiss::uint8 *>(&x), sizeof(x));
+              return x;
+            };
+            auto write = [this](size_t address, etiss::uint64 word, etiss::int32& return_code) {
+              return_code = dbus_access<true>(nullptr, static_cast<etiss::uint64>(address), reinterpret_cast<etiss::uint8 *>(&word), sizeof(word));
+            };
+            mem_manipulator_ = std::make_unique<MemoryWordManipulator<etiss::uint64>>(read, write);
+        }
+        else
+        {
+            etiss::log(etiss::FATALERROR, std::string("Failed to initiliaze MemStack: Architecture bit width not set."));
+        }
+
+        getStruct()->applyCustomAction =  [this](const etiss::fault::Fault &fault, const etiss::fault::Action &action, std::string &errormsg) {
+            auto cmd = action.getCommand();
+
+            size_t pos = 0;
+            std::vector<std::string> split_cmd;
+            while ((pos = cmd.find(" ")) != std::string::npos) {
+                split_cmd.push_back(cmd.substr(0, pos));
+                cmd.erase(0, pos + 1);
+            }
+            split_cmd.push_back(cmd);
+            if (split_cmd.size() > 1)
+            {
+                std::stringstream ss;
```

Comment:
std::stoll would make this less verbose and more robust

Reply:
would prefer boost::lexical_cast over std::sto*, but either over std::stringstream. Fixed with https://github.com/tum-ei-eda/etiss/pull/100/commits/459ac537e8f9c19cfb80f33e7fa09dbcb88e7a34